lib/pengine/native.c

```diff
@@ -545,16 +549,13 @@ native_print(resource_t * rsc, const char *pre_text, long options, void *print_d
 
         /* Ignore target role Started, as it is the default anyways
          * (and would also allow a Master to be Master).
-         * Show if current role differs from target role,
-         * or if target role limits our abilities. */
+         * Show if target role limits our abilities. */
         if (target_role_e == RSC_ROLE_STOPPED) {
             flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset, "%sdisabled", flagOffset?", ":"");
             rsc->cluster->disabled_resources++;
 
         } else if (uber_parent(rsc)->variant == pe_master
-                   && target_role_e > RSC_ROLE_STOPPED
-                   && target_role_e < RSC_ROLE_MASTER
-                   && safe_str_neq(target_role, role2text(rsc->role))) {
+                   && target_role_e == RSC_ROLE_SLAVE) {
```

Comment:
same: only target-role:slave prevents promotion, none, started, default, and master are all equivalent for m/s resources.